src/main/kotlin/io/github/enyason/predictions/TaskExt.kt

```diff
@@ -0,0 +1,55 @@
+import io.github.enyason.domain.models.Prediction
+import io.github.enyason.domain.models.PredictionStatus.*
+import io.github.enyason.io.github.enyason.predictions.Task
+import io.github.enyason.predictions.PredictionsApi
+import kotlinx.coroutines.delay
+import kotlin.coroutines.cancellation.CancellationException
+
+/**
+ * Calling [await] on a [Task] polls <a href="https://replicate.com">Replicate</a> until the prediction status becomes
+ * succeeded. See [io.github.enyason.domain.models.PredictionStatus.SUCCEEDED]
+ * [delayInMillis] is the time delay in milliseconds before another request to the server is triggered
+ */
+suspend fun Task<Prediction>.await(delayInMillis: Long = 2000): Prediction {
```

Comment:
Can we pass in this delayInMillis from the config instead?

Reply:
That makes sense, I will try that out.